script.js

```diff
@@ -1,12 +1,177 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+let allEpisodes;
+let allTVShows;
+let selectedShow;
+let allGenres;
+
+const searchShowsEl = document.querySelector('#searchShows');
+const searchEpisodesEl = document.querySelector('#searchEpisodes');
+const searchResult = document.querySelector('#search-result');
+const selectEpisodesEl = document.querySelector('#select');
+const selectShowEl = document.querySelector('#TVShows');
+const episodesEl = document.querySelector('#episodes');
+const showsEl = document.querySelector('#tvSeries');
+
+const getEpisodeTitle = obj => 'S' + `${obj.season}`.padStart(2, 0) + 'E' + `${obj.number}`.padStart(2, 0) + ` - ${obj.name}`;
+
+const getEpisodeCard = obj => `<div class="col-md-3 m-1 card">
+  <div class="card-body card-css">
+    <h5 class="card-title text-center">${getEpisodeTitle(obj)}</h5>
+    <hr>
+    <img src="${obj.image.original}" class="card-img-top" alt="${getEpisodeTitle(obj)} image">
+    <p class="card-text">${obj.summary}</p>
+  </div>
+</div>`;
+
+const getShowCard = obj => `<div class="card col-12 col-md-10 m-3 show-card g-0" onclick="fetchShowAndDisplay(${obj.id})">
+    <div class="row g-0">
+      <div class="col-md-3">
+        <img src="${obj.image.original}" alt="${obj.name} image" class="img-fluid show-image"/>
+      </div>
+      <div class="col-md-6">
+        <div class="card-body">
+          <h4 class="card-title">${obj.name}</h4>
+          <p class="summary-p">${obj.summary}</p>
+        </div>
+      </div>
+      <div class="col-md-3">
+        <div class="card-body">
+          <p class="card-text"><strong>Rating:</strong> ${obj.rating.average}</p>
+          <p class="card-text"><strong>Genre:</strong> ${obj.genres}</p>
+          <p class="card-text"><strong>Status:</strong> ${obj.status}</p>
+          <p class="card-text"><strong>Runtime:</strong> ${obj.runtime}</p>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>`;
+
+function fetchAllShowsAndDisplay() {
+  fetch('https://api.tvmaze.com/shows')
+    .then(res => res.json())
+    .then(data => {
+      allTVShows = data;
+      displayAllShows(allTVShows);
+      getAllGenres();
+    })
+    .catch(error => console.log(error))
+}
+
+fetchAllShowsAndDisplay();
+
+function fetchShowAndDisplay(id) {
+  fetch(`https://api.tvmaze.com/shows/${id}/episodes`)
+    .then(res => res.json())
+    .then(data => {
+      allEpisodes = data;
+      displaySelectedShow(allEpisodes);
+    })
+    .catch(error => console.log(error));
+}
```

Comment:
I think some of the shows don't have images. you can sort the error handing here.

Reply:
I solved the image loading problems, everything should be loading now.
thank you for spoting the bugs